File: languages/go/entity/availability/ipfs_loader.go
package availability

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// IpfsLoader represents a loader for IPFS data availability.
type IpfsLoader struct {
	Type proto.DataAvailabilityType
	Args LoaderArgs
}

// NewIpfsLoader creates an IpfsLoader instance with the provided IPFS hash.
func NewIpfsLoader(hash string) IpfsLoader {
	return IpfsLoader{
		Type: proto.DataAvailabilityType_IPFS,
		Args: LoaderArgs{
			hash,
		},
	}
}

func (e IpfsLoader) ToProto() *proto.Loader {
	return &proto.Loader{
		Type: e.Type,
		Args: e.Args.ToProto(),
	}
}

File: languages/js/src/entity/availability/ipfs_loader.ts
import * as proto from "../../bridge/proto/availability_entities";
import { Loader } from "./loader";
import { LoaderArgs } from "./loader_args";

/**
 * Represents a loader for IPFS data availability.
 */
export class IpfsLoader implements Loader {
  type: proto.DataAvailabilityType;
  args: LoaderArgs;

  /**
   * Constructs a IpfsLoader object with the specified parameters.
   * @param id 
   */
  constructor(id: string) {
    this.type = proto.DataAvailabilityType.IPFS;
    this.args = new LoaderArgs(id);
  }

  public toProto(): proto.Loader {
    return proto.Loader.fromPartial({
      type: this.type,
      args: this.args.toProto()
    });
  }
}
